src/persist-txn/src/operator.rs

```diff
@@ -622,18 +610,114 @@ impl DataSubscribe {
 
 #[cfg(test)]
 mod tests {
+    use mz_ore::task::JoinHandleExt;
+    use mz_persist_types::Opaque;
+
     use crate::tests::writer;
     use crate::txns::TxnsHandle;
 
     use super::*;
 
+    struct DataSubscribeTask {
+        tx: std::sync::mpsc::Sender<(Option<u64>, tokio::sync::oneshot::Sender<u64>)>,
```

Comment:
```suggestion
        /// Carries step requests. A `None` timestamp requests one step, a `Some(ts)` requests stepping until we progress beyond `ts`.
        tx: std::sync::mpsc::Sender<(Option<u64>, tokio::sync::oneshot::Sender<u64>)>,
```

I'd forget otherwise and would have to look at the code again. 😅

Reply:
Done!